sync/src/synchronizer/mod.rs

```diff
@@ -357,33 +418,112 @@ impl Synchronizer {
         }
     }
 
-    fn find_blocks_to_fetch(&self, nc: &dyn CKBProtocolContext, ibd: IBDState) {
+    fn find_blocks_to_fetch(&mut self, nc: &dyn CKBProtocolContext, ibd: IBDState) {
+        let tip = self.shared.active_chain().tip_number();
+
+        let disconnect_list = {
+            let mut list = self.shared().state().write_inflight_blocks().prune(tip);
+            if let IBDState::In = ibd {
+                // best known < tip and in IBD state, these node can be disconnect
```

Comment:
In the worst case, only the sync peer has the best known is at least as good as tip. But these peers may have unknown blocks which will become best known block later. Should be keep them connected unless we know there are better peers alternatives?